packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -56,24 +54,22 @@ class WebDavClient {
     return response;
   }
 
-  String _constructPath([final String? path]) => [
-        rootClient.baseURL,
-        webdavBasePath,
-        if (path != null) ...[
-          path,
-        ],
-      ]
-          .map((part) {
-            while (part.startsWith('/')) {
-              part = part.substring(1);
-            }
-            while (part.endsWith('/')) {
-              part = part.substring(0, part.length - 1); // coverage:ignore-line
-            }
-            return part;
-          })
-          .where((final part) => part.isNotEmpty)
-          .join('/');
+  Uri _constructUri([final Uri? path]) {
+    assert(
+      path == null || path.path == '/' || !path.path.startsWith('/') || !path.path.endsWith('/'),
```

Comment:
Why is `/` as path not allowed?

Reply:
I just copied the old behavior. '/' is ollwed but everything else shouldn't start or end with a `/`.

Otherwise we could get a `cloud.example.tld/remote.php/webdav/something/` with a trailing `/`. If this is ok I'll happily simplify the code.

Reply:
I think that was only done to correctly put together the path. I assume your changes will work with leading and trailing `/`?

Reply:
IIRC it does; I'll check this later.